### PR TITLE
patch from Uz that makes some illegal operations on pointers error out

### PR DESCRIPTION
**Pull request by mrdudz**
see: http://www.cc65.org/mailarchive/2015-03/11726.html
and: http://www.cc65.org/mailarchive/2015-03/11734.html

this is NOT ready to be merged yet - at least a few more tests should created. also perhaps someone else should test the patch with some bigger program first :) 

that said i have run the testsuite on it and that works - so chances are the patch actually works as intended.

### CODE REVIEW

**Comment by mrdudz** [comment]
ok here are the missing tests.... in theory it can be merged, i guess. perhaps you want to compile contiki with it first, or something :)

File: src/cc65/expr.c [hunk 1]
```diff
@@ -2775,10 +2816,10 @@ static void hie6 (ExprDesc* Expr)
...
     static const GenDesc hie6_ops [] = {
-        { TOK_LT,       GEN_NOPUSH,     g_lt    },
-        { TOK_LE,       GEN_NOPUSH,     g_le    },
-        { TOK_GE,       GEN_NOPUSH,     g_ge    },
-        { TOK_GT,       GEN_NOPUSH,     g_gt    },
+        { TOK_LT,       GEN_NOPUSH | GEN_NOFUNC,     g_lt    },
+        { TOK_LE,       GEN_NOPUSH | GEN_NOFUNC,     g_le    },
+        { TOK_GE,       GEN_NOPUSH | GEN_NOFUNC,     g_ge    },
+        { TOK_GT,       GEN_NOPUSH | GEN_NOFUNC,     g_gt    },
         { TOK_INVALID,  0,              0       }
```
Comment (oliverschmidt) [inline src/cc65/expr.c]: Uz' original patch contains alignment adjustment of this line ;-)

**Comment by oliverschmidt** [comment]
Contiki compiles still fine - I'd opt to merge if no one opposes.

File: src/cc65/expr.c (hunk 1 above)
Comment (mrdudz) [inline src/cc65/expr.c]: perhaps... i had to apply most of it manually :)

**Comment by mrdudz** [comment]
same for the chameleon menu ... go for it :)

File: src/cc65/expr.c (hunk 1 above)
Comment (oliverschmidt) [inline src/cc65/expr.c]: That's why. I fixed it for you - just for once ;-)

...And thanks for preparing the PR incl. additional tests !

**Pull request merged**